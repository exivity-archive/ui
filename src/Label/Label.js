import PropTypes from 'prop-types'
import { Label } from 'reakit'

Label.propTypes = {
  ...Label.propTypes,
  children: PropTypes.node
}

Label.defaultProps = {}

Label.displayName = 'Label'

export default Label
