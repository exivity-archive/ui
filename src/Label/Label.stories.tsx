import React from 'react'
import { Label } from '.'

// @ts-ignore
import { storiesOf } from '@storybook/react'

export default storiesOf('atoms|Label', module)
  .add('default', () => <Label>Default label</Label>)
  .add('secondary', () => <Label secondary>Secondary label</Label>)
  .add('nested', () => <Label>Primary label<Label>Secondary label</Label></Label>)
