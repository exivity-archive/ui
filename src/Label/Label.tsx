import React from 'react'
import styled, { css } from 'styled-components'
import defaultStyledProps from '../utils/testing/defaultStyledProps'
import { fromTheme, globalFont, StyledProps } from '../utils/theme'

interface LabelProps extends StyledProps {
  secondary?: boolean
}

const secondary = css`
  display: block;
  font-size: ${fromTheme(theme => theme.global.sizes.small)}em;
  color: ${fromTheme(theme => theme.global.textColorMuted)};
`

const Label = styled.label <LabelProps>`
  ${globalFont}

  ${props => props.secondary && secondary}

  & > & {
    ${secondary}
  }
`

Label.defaultProps = defaultStyledProps

export default Label
