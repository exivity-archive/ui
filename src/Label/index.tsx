import * as React from 'react'
import styled from 'styled-components'
import theme, { Theme } from '../theme'

const SubLabelWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 33%;
  min-width: 250px;
`

const TopLabel = styled.span`
  font-weight: 600;
  color: #444444;
`
interface ISublabelProps {
  theme: Theme
}

const SubLabel = styled.div`
  margin-top: 5px;
  font-size: 12px;
  color: ${(p: ISublabelProps) => p.theme.colors.gray};
`

const LabelContent = styled.div`
  padding-left: 40px;
`

interface ILabelProps {
  name: string
  description?: string
  className?: string
}

const Label: React.FC<ILabelProps> = ({ name, description, children, className }) => (
  <div className={className}>
    <SubLabelWrapper>
      <TopLabel>{name}</TopLabel>
      <SubLabel theme={theme}>{description}</SubLabel>
    </SubLabelWrapper>
    <LabelContent>
    {children}
    </LabelContent>
  </div>
)

export default styled(Label)`
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;
  width: 100%;
  margin: 40px 0;
`
