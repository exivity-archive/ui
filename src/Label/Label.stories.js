import React from 'react'
import { storiesOf } from '@storybook/react'

import Label from './Label'

storiesOf('atoms|Label', module)
  .add('default', () => <Label>I'm a label</Label>)
