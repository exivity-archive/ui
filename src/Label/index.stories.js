import React from 'react'
import Label from '../Label'

import { storiesOf } from '@storybook/react'
import { CenterDecorator } from '../utils/testing/decorators/CenterDecorator'

export default storiesOf('Label', module)
  .addDecorator(CenterDecorator)
  .add('Label', () => {
    return <Label name='Label name' description='Label description' />
  })
