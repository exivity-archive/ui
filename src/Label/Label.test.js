/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import Label from './Label'

test('renders default Label', () => {
  const component = renderer.create(<Label />)
  expect(component.toJSON()).toMatchSnapshot()
})
