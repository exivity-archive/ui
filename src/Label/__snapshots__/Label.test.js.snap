// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Label 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  display: inline-block;
}

<label
  className="c0 c1"
/>
`;
